roles/download_file/tasks/main.yml

```
@@ -4,3 +4,13 @@
   ansible.builtin.get_url:
     url: "{{ download_file_url }}"
     dest: "{{ download_file_destination }}"
+  when: download_file_locally == false
+
+# tasks file for download_file
+- name: Download file locally
+  ansible.builtin.get_url:
+    url: "{{ download_file_url }}"
+    dest: "{{ download_file_destination }}"
+  delegate_to: 127.0.0.1
```

can't we do something like this pseudocode?

```
delegate_to: {{ download_file_locally && '127.0.0.1' || omit() }}
```

(i didn't check the docs, but this feels like it could be simplified to the same task)

---

i think this might be an approach to consider:
```suggestion
  delegate_to: {{ download_file_delegate_to | default(omit) }}
```

it would make different readmes, but i think it the log output contains the delegated host anyways